addons/purchase/invoice.py

```diff
@@ -35,8 +35,8 @@ def _prepare_invoice_line_from_po_line(self, line):
             qty = line.product_qty - line.qty_invoiced
         else:
             qty = line.qty_received - line.qty_invoiced
-        if float_compare(qty, 0.0, precision_rounding=line.product_uom.rounding) <= 0:
-            # If received quantities or quantities is zero is better not create invoice lines
+        if float_compare(qty, 0.0,
+                         precision_rounding=line.product_uom.rounding) == 0:
             return {}
```

Comment:
@jbeficent @pedrobaeza We can return as original method the dictionary with qty=0, then in this line https://github.com/OCA/OCB/blob/9.0/addons/purchase/invoice.py#L76 make the condition like this:
if not data.get('quantity', 0.0):
.....
by this way you can modify the units as you want and leave this method as the original.
@jbeficent  Do you want that I make the PR??